src/app/mork/tests/api/v1/routers/test_users.py

```diff
@@ -240,6 +240,60 @@ async def test_users_read_filter(
     assert len(total_requests) == 5
 
 
+@pytest.mark.anyio
+@pytest.mark.parametrize(
+    "params, expected_count",
+    [
+        ({"email": "TestUser@Example.COM"}, 1),
+        ({"username": "TestUser123"}, 1),
+        ({"email": "TestUser@Example.COM", "username": "TestUser123"}, 1),
+        ({"email": "TestUser@Example.COM", "service": "ashley"}, 1),
+        ({"username": "TestUser123", "deletion_status": "to_delete"}, 1),
+        # Case insensitive search tests
+        ({"email": "testuser@example.com"}, 1),  # Different case
+        ({"username": "testuser123"}, 1),  # Different case
+    ],
+)
+async def test_users_read_filter_email_username(
+    db_session,
+    http_client: AsyncClient,
+    auth_headers: dict,
+    params: dict,
+    expected_count: int,
+):
+    """Test scenarios with email and username query parameters when retrieving users."""
+    UserServiceStatusFactory._meta.sqlalchemy_session = db_session
+    UserFactory._meta.sqlalchemy_session = db_session
+
+    # Create specific users for email and username tests
+    UserFactory.create(
```

Comment:
You should probably create at least one additional user to make sure the filtering logic is working, and that it is not just displaying all users, i.e. one.